jobbergate-cli/jobbergate_cli/subapps/clusters/tools.py

```diff
@@ -0,0 +1,111 @@
+"""
+Provide tool functions for working with Cluster data
+"""
+
+import json
+from datetime import datetime
+from typing import Dict, List, Optional, cast
+
+from loguru import logger
+
+from jobbergate_cli.config import settings
+from jobbergate_cli.exceptions import Abort
+from jobbergate_cli.requests import make_request
+from jobbergate_cli.schemas import ClusterCacheData, JobbergateContext
+from jobbergate_cli.text_tools import conjoin
+
+
+def pull_cluster_names_from_api(ctx: JobbergateContext) -> List[str]:
+    assert ctx.client is not None
+
+    response_data = cast(
+        Dict,
+        make_request(
+            ctx.client,
+            "/cluster/graphql/query",
+            "POST",
+            expected_status=200,
+            abort_message="There was a problem retrieving registered clusters from the Cluster API",
+            abort_subject="COULD NOT RETRIEVE CLUSTERS",
+            support=True,
+            json=dict(
+                query="query {cluster{clientId}}",
+                variables=dict(),
+            ),
+        ),
+    )
+
+    try:
+        cluster_names = [e["clientId"] for e in response_data["data"]["cluster"]]
```

Comment:
there are difference between the client ID and the name. They both exist in the Cluster API and they are different. Both are unique. So, you can search in the GraphQL query by the `clientId` and ask for the `name` field

Reply:
Adjusted in commit 5af24fc